File: modules/core/app/model/matching/StringMatcher.java
package model.matching;

import java.util.List;

public class StringMatcher extends Matcher<String, StringMatch, StringMatchingResult> {

  public StringMatcher() {
    super((str1, str2) -> str1.equals(str2), StringMatch::new);
  }

  @Override
  protected StringMatchingResult instantiateMatch(List<StringMatch> matches, List<String> firstCollection,
      List<String> secondCollection) {
    return new StringMatchingResult(matches, firstCollection, secondCollection);
  }

}

File: app/model/matching/StringMatcher.scala
package model.matching

final case class StringMatch(matchType: MatchType, userArg: String, sampleArg: String) extends Match[String]

object StringMatcher extends Matcher[String, StringMatch] {

  type StringMatchingResult = MatchingResult[String, StringMatch]

  override protected def instantiateMatch(ua: String, sa: String): StringMatch =
    StringMatch(MatchType.SUCCESSFUL_MATCH, ua, sa)

}
